### DISCUSSION TITLE
Create branch or checkout in all submodules

### DISCUSSION THREAD

**Discussion by UgaitzEtxebarria** [gitextensions/gitextensions#9449] [Q&A] [answered]
Hello,

Is there any way to create a branch in every submodules of the repo?

And to checkout?

Thank you!

**Comment by RussKie** [comment, marked as answer]
Not via the UI. But if this something you'd want to do on a regular basis you could write a script, and hook up to specific events. Check out Settings > Git Extensions > Scripts.

**Reply by UgaitzEtxebarria** [reply]
I made it using a script like this:
Command: git submodule foreach
Arguments: git checkout -b {UserInput} 

Thanks!
